mcstatus/status_response.py

```diff
@@ -0,0 +1,317 @@
+        "gold": "6",
+        "yellow": "e",
+        "dark_green": "2",
+        "green": "a",
+        "aqua": "b",
+        "dark_aqua": "3",
+        "dark_blue": "1",
+        "blue": "9",
+        "light_purple": "d",
+        "dark_purple": "5",
+        "white": "f",
+        "gray": "7",
+        "dark_gray": "8",
+        "black": "0",
+    },
+}
+
+
+def _validate_data(raw: Dict[str, Any], who: str, required: Iterable[Tuple[str, type]]) -> None:
+    """Ensure that all required keys are present, and have the specified type.
+
+    :param raw: The raw dict answer to check.
+    :param who: The name of the object that is checking the data. Example `status`, `player` etc.
+    :param required: An iterable of string and type. The string is the required key which must be in `raw`, and
+        the `type` is the type that the key must be. If you want to ignore check of the type, set the type to `object`.
+    :raises ValueError: If the required keys are not present.
+    :raises TypeError: If the required keys are not of the specified type.
+    """
+    for required_key, required_type in required:
+        if required_key not in raw:
+            raise ValueError(f"Invalid {who} object (no '{required_key}' value)")
+        if not isinstance(raw[required_key], required_type):
+            raise TypeError(
+                f"Invalid {who} object (expected '{required_key}' to be {required_type}, was {type(raw[required_key])})"
+            )
+
+
+@dataclass
+class AbstractDataclass(ABC):
+    """Abstract base class for all dataclasses.
+
+    We can't use just `ABC` because we need raise TypeError if we try to
+    instantiate an abstract dataclass. Class which inherit from `ABC` raises
+    it only if it has abstract methods.
+    """
+
+    def __new__(cls, *args, **kwargs):
+        if cls is AbstractDataclass or cls.__bases__[0] is AbstractDataclass:
+            raise TypeError("Cannot instantiate abstract class.")
+        return super().__new__(cls)
+
+
+@dataclass
+class MCStatusResponse(AbstractDataclass):
+    """Class for storing shared data from a status response.
+
+    :param motd: Message Of The Day. Also known as `Description`.
+    :param latency: Latency between a server and the client (you). In milliseconds.
+    """
+
+    players: StatusPlayers
+    version: StatusVersion
+    motd: str
+    latency: float
+
+    @classmethod
+    @abstractmethod
+    def build(cls, *args, **kwargs) -> MCStatusResponse:
+        """Build MCStatusResponse and check is it valid.
+
+        :param args: Arguments in specific realisation.
+        :param kwargs: Keyword arguments in specific realisation.
+        :return: `MCStatusResponse` object.
+        """
+        raise NotImplementedError("You can't use abstract methods.")
+
+
+@dataclass
+class JavaStatusResponse(MCStatusResponse):
+    """Class for storing JavaServer data from a status response.
+
+    :param icon: Icon of the server. Can be unset. BASE64 encoded.
```

Comment:
Small word order improvement
```suggestion
    :param icon: Base64 encoded icon of the server. Can be unset.
```